Core/include/Acts/Vertexing/KalmanVertexUpdater.hpp

```diff
@@ -18,72 +18,94 @@ namespace KalmanVertexUpdater {
-  SquareMatrix3 momWeightInv = SquareMatrix3::Zero();
+  // W_k
+  SquareMatrix3 wMat = SquareMatrix3::Zero();
 };
 
 /// @brief Updates vertex with knowledge of new track
-/// @note KalmanVertexUpdater updates the vertex w.r.t. the
-/// newly given track, but does NOT add the track to the
-/// TrackAtVertex list. Has to be done manually after calling
-/// this method
+/// @note KalmanVertexUpdater updates the vertex when trk is added to the fit.
+/// However, it does not add the track to the TrackAtVertex list. This to be
+/// done manually after calling the method.
 ///
 /// @param vtx Vertex to be updated
 /// @param trk Track to be used for updating the vertex
 template <typename input_track_t>
 void updateVertexWithTrack(Vertex<input_track_t>& vtx,
                            TrackAtVertex<input_track_t>& trk);
 
-/// @brief Updates vertex position
-///
-/// @param vtx Old vertex
-/// @param linTrack Linearized version of track to be added or removed
+/// @brief Calculates updated vertex position and covariance as well as the
+/// updated track momentum when adding/removing linTrack. Saves the result in
+/// cache.
+/// @param vtx Vertex
+/// @param linTrack Linearized track to be added or removed
 /// @param trackWeight Track weight
 /// @param sign +1 (add track) or -1 (remove track)
-/// @param[out] matrixCache A cache to store matrix information
+/// @note Tracks are removed during the smoothing procedure to compute
+/// the chi2 of the track wrt the updated vertex position
+/// @param[out] cache A cache to store the results of this function
 template <typename input_track_t>
-void updatePosition(const Acts::Vertex<input_track_t>& vtx,
-                    const Acts::LinearizedTrack& linTrack, double trackWeight,
-                    int sign, MatrixCache& matrixCache);
+void calculateUpdate(const Acts::Vertex<input_track_t>& vtx,
+                     const Acts::LinearizedTrack& linTrack,
+                     const double trackWeight, const int sign, Cache& cache);
 
 namespace detail {
 
-/// @brief Takes old and new vtx and calculates position chi2
+/// @brief Calculates the update of the vertex position chi2 after
+/// adding/removing the track
 ///
-/// @param oldVtx Old vertex
-/// @param matrixCache A cache to store matrix information
+/// @param oldVtx Vertex before the track was added/removed
+/// @param cache Cache containing updated vertex position
 ///
 /// @return Chi2
 template <typename input_track_t>
-double vertexPositionChi2(const Vertex<input_track_t>& oldVtx,
-                          const MatrixCache& matrixCache);
+double vertexPositionChi2Update(const Vertex<input_track_t>& oldVtx,
+                                const Cache& cache);
 
 /// @brief Calculates chi2 of refitted track parameters
 /// w.r.t. updated vertex
 ///
 /// @param linTrack Linearized version of track
-/// @param matrixCache A cache to store matrix information
+/// @param cache Cache containing some quantities needed in
```

Comment:
Could you provide a more informative description here? I don't fully understand it 😅

Reply:
When we call `updateVertexWithTrack`, we execute the function `calculateUpdate`. The latter fills `cache`. The quantities are saved because we need them again later, for example to compute the chi2 of the track parameters like in the function below :)